index.js

```diff
@@ -48,7 +58,14 @@ async function main(adapterAddress, startBlock = 0, endBlock = Infinity) {
     if (to !== ethers.constants.AddressZero) {
       if (scores[to]) {
         const { prevCheckpoint, score } = scores[to];
-        scores[to].score = score.plus(prevCheckpoint.amount.times(block - prevCheckpoint.block));
+        if (startBlock > prevCheckpoint.block) {
+          if (startBlock < block) {
+            scores[to].score = score.plus(prevCheckpoint.amount.times(block - startBlock));
+          }
+        } else {
+          scores[to].score = score.plus(prevCheckpoint.amount.times(block - prevCheckpoint.block));
+        }
```

Comment:
I find this section really hard to reason about! Could you please write some comments?